server/src/main/scala/de/frosner/broccoli/models/ParameterValue.scala

```diff
@@ -0,0 +1,92 @@
+package de.frosner.broccoli.models
+
+import com.typesafe.config.{Config, ConfigValue, ConfigValueType}
+import com.typesafe.config.impl.ConfigInt
+import de.frosner.broccoli.services.{ParameterValueParsingException, TemplateParameterNotFoundException}
+import org.apache.commons.lang3.StringEscapeUtils
+import play.api.libs.json._
+
+import scala.util.{Failure, Success, Try}
+
+object ParameterValue {
+
+  implicit val parameterValueWrites: Writes[ParameterValue] = new Writes[ParameterValue] {
+    override def writes(paramValue: ParameterValue) = paramValue.asJsValue
+  }
+
+  def constructParameterValueFromTypesafeConfig(parameterType: ParameterType,
+                                                configValue: ConfigValue): Try[ParameterValue] =
+    Try {
+      parameterType match {
+        case ParameterType.Raw =>
+          RawParameterValue(configValue.unwrapped().toString)
+        case ParameterType.String =>
+          StringParameterValue(configValue.unwrapped.asInstanceOf[String])
+        case ParameterType.Integer =>
+          IntParameterValue(configValue.unwrapped().asInstanceOf[Int])
+        case ParameterType.Decimal =>
+          DecimalParameterValue(configValue.unwrapped().asInstanceOf[BigDecimal])
+      }
+    }
+
+  def constructParameterValueFromJson(parameterName: String,
+                                      template: Template,
+                                      jsValue: JsValue): Try[ParameterValue] =
+    Try {
+      val parameterInfo =
+        template.parameterInfos.getOrElse(parameterName, ParameterInfo(parameterName, None, None, None, None, None))
+      constructParameterValueFromJson(parameterInfo.`type`.getOrElse(ParameterType.Raw), jsValue) match {
+        case Some(param) => param
+        case None =>
+          throw ParameterValueParsingException(template.id, parameterName)
+      }
+    }
+
+  def constructParameterValueFromJson(parameterType: ParameterType, jsValue: JsValue): Option[ParameterValue] =
+    Try {
+      parameterType match {
+        case ParameterType.Raw =>
+          RawParameterValue(jsValue.as[String])
+        case ParameterType.String =>
+          StringParameterValue(jsValue.as[String])
+        case ParameterType.Integer =>
+          IntParameterValue(jsValue.as[Int])
+        case ParameterType.Decimal =>
+          DecimalParameterValue(jsValue.as[BigDecimal])
+      }
+    } match {
+      case Success(s) => Some(s)
+      case Failure(_) => None
+    }
+}
+sealed trait ParameterValue {
+  def isEmpty: Boolean
```

Comment:
Why do we need `isEmpty`?

Reply:
Because we filter on empty parameters in the [`addInstance`](https://github.com/FRosner/cluster-broccoli/blob/master/server/src/main/scala/de/frosner/broccoli/services/InstanceService.scala#L209) and [`updateInstance`](https://github.com/FRosner/cluster-broccoli/blob/master/server/src/main/scala/de/frosner/broccoli/services/InstanceService.scala#L245) methods in the `InstanceService`. Since this only made sense for String to me I defaulted the others to true.

Reply:
I think we should refactor this. If we require this check maybe we can change the semantics of adding instances. If you want we can have a quick call to discuss that.